webapp/channels/src/components/announcement_bar/notification_permission_bar/index.tsx

```diff
@@ -10,26 +10,26 @@ import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
 import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
 
 import {AnnouncementBarTypes} from 'utils/constants';
+import {requestNotificationPermission, isNotificationAPISupported} from 'utils/notifications';
 
 export default function NotificationPermissionBar() {
     const isLoggedIn = Boolean(useSelector(getCurrentUserId));
 
-    const [show, setShow] = useState(Notification.permission === 'default');
+    const [show, setShow] = useState(isNotificationAPISupported() ? Notification.permission === 'default' : false);
 
-    const handleClose = useCallback(() => {
-        // If the user closes the bar, don't show the notification bar any more for the rest of the session, but
-        // show it again after refresh.
-        setShow(false);
-    }, []);
+    if (!show || !isLoggedIn || !isNotificationAPISupported()) {
+        return null;
+    }
 
-    const handleClick = useCallback(() => {
-        Notification.requestPermission().then(() => {
-            setShow(false);
-        });
-    }, []);
+    async function handleClick() {
```

Comment:
Could you change these back to `useCallback`s? I think that's important here because the `AnnouncementBar` component is decent sized component

Reply:
changed back